config/v1/types_infrastructure.go

```diff
@@ -895,6 +951,21 @@ type VSpherePlatformStatus struct {
 	// datacenter DNS, a DNS service is hosted as a static pod to serve those hostnames
 	// to the nodes in the cluster.
 	NodeDNSIP string `json:"nodeDNSIP,omitempty"`
+
+	// loadBalancerType defines if an user-managed or openshift-managed loadbalancer
+	// is to be used for the openshift API and Ingress endpoints.
+	// When set to OpenShiftManaged the static pods in charge of API and Ingress traffic load-balancing
+	// defined in the machine config operator will be deployed.
+	// When set to UserManaged these static pods will not be deployed and it is expected that
+	// the load balancer is configured out of band by the deployer.
+	// When omitted, this means no opinion and the platform is left to choose a reasonable default
+	// The current default value is OpenShiftManaged.
+	// +kubebuilder:default:="OpenShiftManaged"
+	// +kubebuilder:validation:Enum:="OpenShiftManaged";"UserManaged"
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="loadBalancerType is immutable"
```

Comment:
On an upgraded cluster, this value will be empty when read (oldSelf) and then the next update will have the default value (self).  How will this expression allow a zero-diff write from the client perspective to succeed?

Reply:
Maybe both reads see the default value?